### ISSUE TITLE
Rule Change: prefer-regex-literals should not report an error if the regex is used multiple times

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#17562]
### What rule do you want to change?

prefer-regex-literals

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
const regex = new RegExp( '\w+(?![a-z])' );
for ( const value of data ) {
	if ( regex.test( value ) === true ) {
		result += value.replace( regex, 'foo' );
		continue;
	}

	result += 'something else';
}
```

### What does the rule currently do for this code?

Report an error

### What will the rule do after it's changed?

No error

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

When a RegExp is used multiple times it should not report an error for prefer-regex-literals, as it would mean that the regex code has to be duplicated which makes maintenance more work and also risk introducing bugs in case only 1 regex is changed but the other one is forgotten.

**Comment by mdjermanovic** [comment]
Hi @kkmuffme, thanks for the issue! 

> When a RegExp is used multiple times it should not report an error for prefer-regex-literals, as it would mean that the regex code has to be duplicated which makes maintenance more work and also risk introducing bugs in case only 1 regex is changed but the other one is forgotten.

In the example from the original post, regex code doesn't have to be duplicated. The rule just suggests replacing `new RegExp()` with a regex literal:

```js
const regex = /w+(?![a-z])/;
for ( const value of data ) {
	if ( regex.test( value ) === true ) {
		result += value.replace( regex, 'foo' );
		continue;
	}

	result += 'something else';
}
```

(note that this is equivalent to the original code, but the original code probably has a bug in `\w`, which evaluates to just `w`).

I think whether the regex is used once or multiple times should not affect the logic of this rule.
